From c9ff5bb1ce8eb4fa61d62b688a04637a98590010 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 03 Jan 2021 23:43:33 +0200
Subject: [PATCH] [SW-28128] gaudi: security: refactor lbw pcie rr

Move the preboot_secure_pcie_rr() function calls inside the
preboot_secure_lbw_rr() function.
This spares the some code and defines a point for common LBW RRs
configurations.

Change-Id: Id521ef35a1ff2b55eb7dc5fae458bd2b6a721f89
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index e233878..d5ab1e4 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1114,6 +1114,7 @@
 static void preboot_secure_lbw_rr(int rr, u64 min, u64 max)
 {
 	preboot_secure_lbw_rr_gen(rr, min, max, IF_RTR);
+	preboot_secure_pcie_rr(rr, min, max);
 }
 
 #define SYNC_MNGR_DIST		0x20000
@@ -1133,13 +1134,11 @@
 	min = mmSRAM_Y0_X0_BANK_BASE - 1;
 	max = mmSIF_RTR_0_BASE;
 	preboot_secure_lbw_rr(0, min, max);
-	preboot_secure_pcie_rr(0, min, max);
 
 	/* CPU_TIMESTAMP: RR_1 */
 	min = mmCPU_TIMESTAMP_BASE - 1;
 	max = mmDMA_IF_W_S_BASE;
 	preboot_secure_lbw_rr(1, min, max);
-	preboot_secure_pcie_rr(1, min, max);
 
 	/* SYNC_MNGR: RR_2 up to RR_5 */
 	for (i = 0; i < 4; i++) {
@@ -1153,55 +1152,46 @@
 	min = mmHBM0_BASE - 1;
 	max = HL_CFG_BASE + mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR;
 	preboot_secure_lbw_rr(6, min, max);
-	preboot_secure_pcie_rr(6, min, max);
 
 	/* GIC_H up to PCIE_WRAP: RR_7 */
 	min = max + sizeof(u32) - 1;
 	max = mmPCIE_WRAP_BASE;
 	preboot_secure_lbw_rr(7, min, max);
-	preboot_secure_pcie_rr(7, min, max);
 
 	/* PCIE_CORE up to PCIE_CORE_L: RR_8 */
 	min = mmPCIE_CORE_BASE - 1;
 	max = HL_CFG_BASE + mmPCIE_CORE_MSI_REQ;
 	preboot_secure_lbw_rr(8, min, max);
-	preboot_secure_pcie_rr(8, min, max);
 
 	/* PCIE_CORE_H up to MMU_UP: RR_9 */
 	min = max + sizeof(u32) - 1;
 	max = mmMMU_UP_BASE;
 	preboot_secure_lbw_rr(9, min, max);
-	preboot_secure_pcie_rr(9, min, max);
 
 	/* PSOC_I2C_M0 up to PSOC_TIMESTAMP: RR_10 */
 	min = mmPSOC_I2C_M0_BASE - 1;
 	max = mmPSOC_TIMESTAMP_BASE;
 	preboot_secure_lbw_rr(10, min, max);
-	preboot_secure_pcie_rr(10, min, max);
 
 	/* PSOC_GPIO0 up to NIC0_MAC_CH0: RR_11 */
 	min = mmPSOC_GPIO0_BASE - 1;
 	max = mmNIC0_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(11, min, max);
-	preboot_secure_pcie_rr(11, min, max);
 
 	/* NIC0_TS up to NIC1_MAC_CH0: RR_12 */
 	min = mmNIC0_TS_BASE - 1;
 	max = mmNIC1_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(12, min, max);
-	preboot_secure_pcie_rr(12, min, max);
 
 	/* NIC1_TS up to NIC2_MAC_CH0: RR_13 */
 	min = mmNIC1_TS_BASE - 1;
 	max = mmNIC2_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(13, min, max);
-	preboot_secure_pcie_rr(13, min, max);
 
 	/* NIC3_TS up to NIC4_MAC_CH0: RR_14 */
 	min = mmNIC3_TS_BASE - 1;
 	max = mmNIC4_MAC_CH0_BASE;
 	preboot_secure_lbw_rr(14, min, max);
-	preboot_secure_pcie_rr(14, min, max);
 }
 
 static void preboot_secure_hbw_rr(int rr, u64 base, u64 mask)
